daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala

```diff
@@ -46,6 +46,42 @@ sealed abstract class Value[+Cid] extends Product with Serializable {
       case ValueMap(x) => ValueMap(x.mapValue(_.mapContractId(f)))
     }
 
+  // TODO (Nick Smith) - Still not tail-recursive
+  def mapContractIdOpt[Cid2](f: Cid => Option[Cid2]): Option[Value[Cid2]] = {
```

Comment:
```suggestion
  def collectContractIdOpt[Cid2](f: Cid => Option[Cid2]): Option[Value[Cid2]] = {
```

I've given this function its more Scala-aligned name, but I don't see where you use it so I'm not even sure it has the right semantics.